--- ckermit-8.0.211/ckcasc.h~	2004-01-07 19:16:20.000000000 +0100
+++ ckermit-8.0.211/ckcasc.h	2007-04-01 01:38:09.000000000 +0200
@@ -42,7 +42,7 @@
 #define SYN   22        /* SYN, Ctrl-V */
 #define ETB   23        /* Ctrl-W */
 #define CAN   24        /* CAN, Ctrl-X */
-#define EM    25        /* Ctrl-Y */
+#define __EM  25        /* Ctrl-Y */
 #define SUB   26        /* SUB Ctrl-Z */
 #define ESC   27        /* Escape Ctrl-[ */
 #define XFS   28        /* Field Separator,  Ctrl-Backslash */
