diff -urNp -x '*.orig' ckermit-9.0.302.org/makefile ckermit-9.0.302/makefile
--- ckermit-9.0.302.org/makefile	2011-08-21 17:12:07.000000000 +0200
+++ ckermit-9.0.302/makefile	2021-10-22 22:35:06.702445806 +0200
@@ -6834,6 +6834,45 @@ linux+krb5+openssl+zlib+shadow+pam:
 	-lm -lncurses -ltermcap -lssl -lcrypto -lgssapi_krb5 \
 	-lkrb5 -lcom_err -lk5crypto -lcrypt -lresolv -lpam -ldl -lz"
 
+linux-PLD+krb5+openssl+zlib+pam+shadow:
+	@echo 'Making C-Kermit $(CKVER) for Linux on i386 with PAM,KRB5,SSL...'
+	$(MAKE) xermit KTARGET=$${KTARGET:-$(@)} "CC = $(CC)" "CC2 = $(CC)" \
+	"CFLAGS = $(OPT) -funsigned-char -pipe -DPOSIX -DLINUX -DNOCOTFMC \
+	-DCK_AUTHENTICATION -DCK_KERBEROS -DKRB5 -DCK_SHADOW -DOPENSSL_097 \
+	-DCK_ENCRYPTION -DCK_CAST -DCK_DES -DLIBDES -DCK_SSL -DCK_PAM -DZLIB \
+	-DCK_CURSES -DCK_POSIX_SIG -DTCPSOCKET -DLINUXFSSTND -DHAVE_CRYPT_H \
+	\
+	-I/usr/include/security -I/usr/include/openssl -I/usr/include/ncurses \
+	-I/usr/include/krb5 -I/usr/include/et $(KFLAGS)" "LNKFLAGS = $(LNKFLAGS)" \
+	"LIBS = -lm -lncurses -lssl -lcrypto -lgssapi_krb5 -lutil \
+	-lkrb5 -lcom_err -lk5crypto -lcrypt -lresolv -lpam -ldl -lz -ltinfo"
+
+linux-PLD+krb5heimdal+openssl+zlib+pam+shadow:
+	@echo 'Making C-Kermit $(CKVER) for Linux on i386 with PAM,KRB5,SSL...'
+	$(MAKE) xermit KTARGET=$${KTARGET:-$(@)} "CC = $(CC)" "CC2 = $(CC)" \
+	"CFLAGS = $(OPT) -funsigned-char -pipe -DPOSIX -DLINUX -DNOCOTFMC \
+	-DCK_AUTHENTICATION -DCK_KERBEROS -DKRB5 -DCK_SHADOW -DOPENSSL_097 \
+	-DCK_ENCRYPTION -DCK_CAST -DCK_DES -DLIBDES -DCK_SSL -DCK_PAM -DZLIB \
+	-DCK_CURSES -DCK_POSIX_SIG -DTCPSOCKET -DLINUXFSSTND -DHAVE_CRYPT_H \
+	-DHEIMDAL \
+	-I/usr/include/security -I/usr/include/openssl -I/usr/include/ncurses \
+	-I/usr/include/krb5 -I/usr/include/et $(KFLAGS)" "LNKFLAGS = $(LNKFLAGS)" \
+	"LIBS = -lm -lncurses -lssl -lcrypto -lgssapi -lutil \
+	-lkrb5 -lcom_err -lcrypt -lresolv -lpam -ldl -lz -ltinfo"
+
+linux-PLD+openssl+zlib+pam+shadow:
+	@echo 'Making C-Kermit $(CKVER) for Linux on i386 with PAM,KRB5,SSL...'
+	$(MAKE) xermit KTARGET=$${KTARGET:-$(@)} "CC = $(CC)" "CC2 = $(CC)" \
+	"CFLAGS = $(OPT) -funsigned-char -pipe -DPOSIX -DLINUX -DNOCOTFMC \
+	-DCK_AUTHENTICATION -DCK_SHADOW -DOPENSSL_097 \
+	-DCK_ENCRYPTION -DCK_CAST -DCK_DES -DLIBDES -DCK_SSL -DCK_PAM -DZLIB \
+	-DCK_CURSES -DCK_POSIX_SIG -DTCPSOCKET -DLINUXFSSTND -DHAVE_CRYPT_H \
+	\
+	-I/usr/include/security -I/usr/include/openssl -I/usr/include/ncurses \
+	-I/usr/include/krb5 -I/usr/include/et $(KFLAGS)" "LNKFLAGS = $(LNKFLAGS)" \
+	"LIBS = -lm -lncurses -lssl -lcrypto -lutil \
+	-lcrypt -lresolv -lpam -ldl -lz -ltinfo"
+
 linuxnotcp:
 	$(MAKE) linux KTARGET=$${KTARGET:-$(@)} "KFLAGS = -DNONET $(KFLAGS)"
 
